### PR TITLE
Upgrade Frontend dependencies

### PR DESCRIPTION
**Pull request by Dhwaniartefact**
Frontend dependencies are upgraded by removing yarn packages and installing npm version 9. 

`overrides` feature of npm is used to upgrade specific version of `devDependencies` and `dependencies`. 

Reference : https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides

### CODE REVIEW

File: hack/Dockerfile
Comment (replaceafill) [inline hack/Dockerfile]: Should this and line 138 below use tabs instead of spaces?

File: hack/Dockerfile
Comment (replaceafill) [inline hack/Dockerfile]: I presume this has the same effect of the `yarn ... install ...` line below? If that's the case the output of this command will be owned by `root` and not by the `archivematica` user.

File: src/dashboard/frontend/package.json
Comment (replaceafill) [inline src/dashboard/frontend/package.json]: Indentation seems a bit off?

Comment (replaceafill) [reply src/dashboard/frontend/package.json]: I wonder if we should add this file to the pre-commit hook that formats other files in the repo? https://github.com/artefactual/archivematica/blob/155f229fa7a15cc7f01a57eb5a9fc0335a20f5c8/.pre-commit-config.yaml#L7-L11

File: hack/Dockerfile
Comment (Dhwaniartefact) [reply hack/Dockerfile]: oops. That's something due to editor. Let me fix it.

File: hack/Dockerfile
Comment (replaceafill) [inline hack/Dockerfile:149]: If this is removed, should we also delete the `yarn.lock` file?

File: src/dashboard/frontend/package.json
Comment (replaceafill) [reply src/dashboard/frontend/package.json]: I tested this and the hook works as expected but I'm hesitant about its default behavior of sorting mapping keys. It also has a couple of [arguments](https://github.com/pre-commit/pre-commit-hooks#pretty-format-json) that could be useful: `--no-sort-keys` to get the file formatted with no key reordering and `--top-keys` which allows you to define keys you want to keep in place. We could use the latter for the metadata keys of the package such as `name`, `description` and `version`. In any case we can discuss it after this PR gets merged :slightly_smiling_face:

File: hack/Dockerfile
Comment (Dhwaniartefact) [reply hack/Dockerfile]: Thanks and done.

File: hack/Dockerfile
Comment (Dhwaniartefact) [reply hack/Dockerfile:149]: Yes it is removed in next commit. Thanks for reminding me.

File: hack/Dockerfile
Comment (Dhwaniartefact) [reply hack/Dockerfile]: Thanks for suggestions. I have installed npm globally by root user separately.

File: src/dashboard/frontend/package.json
Comment (sevein) [inline src/dashboard/frontend/package.json:64]: I didn't know overrides can do this, cool.

File: hack/Dockerfile
Comment (sevein) [inline hack/Dockerfile:54]: Can we use 18.x or 20.x? Related: https://endoflife.date/nodejs.

File: hack/Dockerfile
Comment (sevein) [inline hack/Dockerfile:140]: Why 9.8.1 and not the latest 9.x, e.g. 9.9.2?

**Review by sevein — APPROVED** [review]
I haven't tested but this looks good to me. I defer to Douglas for the final decision. It would be great if we can use the latest LTS available.

File: hack/Dockerfile
Comment (Dhwaniartefact) [reply hack/Dockerfile:140]: Thank you and noted all your suggestions.  I made changes accordingly in https://github.com/artefactual/archivematica/pull/1875

**Comment by Dhwaniartefact** [comment]
Closing this PR since npm and node are upgraded in https://github.com/artefactual/archivematica/pull/1875

**Pull request closed without merging**
